ignite/src/main/java/org/hibernate/ogm/datastore/ignite/IgniteDialect.java

```diff
@@ -146,12 +145,13 @@ public void insertOrUpdateTuple(EntityKey key, TuplePointer tuplePointer, TupleC
 
 		Object keyObject = null;
 		BinaryObjectBuilder builder = null;
-		IgniteTupleSnapshot tupleSnapshot = (IgniteTupleSnapshot) tuple.getSnapshot();
-		keyObject = tupleSnapshot.getCacheKey();
 		if ( tuple.getSnapshotType() == SnapshotType.UPDATE ) {
-			builder = provider.createBinaryObjectBuilder( tupleSnapshot.getCacheValue() );
+			IgniteTupleSnapshot tupleSnapshot = (IgniteTupleSnapshot) tuple.getSnapshot();
+			keyObject = tupleSnapshot.getCacheKey();
+			builder = provider.createBinaryObjectBuilder( entityCache.get( keyObject ) );
```

Comment:
I think that you replace my new code with old vesrion. What was wrong in insertOrUpdateTuple?
In any case `entityCache.get( keyObject )` here isn't good.

Reply:
May be.... Where can I take a last version of your code?

Reply:
Do you mean this commit 'https://github.com/Z-z-z-z/hibernate-ogm-ignite/commit/4835b9718b95748623910fe5008b72aee7ed025f'?

Reply:
yes